var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -83,7 +83,7 @@ class Mpich(AutotoolsPackage):
 
     variant(
         'two_level_namespace',
-        default=False,
+        default=(sys.platform == 'darwin'),
```

Comment:
Can we make the variant conditional on `when="platform=linux"` and use something like:
```python
if self.satisfies("+two_level_namespace") or self.satisfies("platform=darwin"):
    ...
```
in the recipe? It would also be fine with me to remove the variant set to False the underlying logic in linux, to True on Darwin.

Reply:
"two-level" vs. "flat" namespaces are a macOS thing. They have no effect elsewhere. Given the Darwin constraint below, I don't think this should be a variant at all.

These two options are similar to the effect of `RTLD_GLOBAL` when opening a shared library, except that the choice is made earlier – one chooses when the library is built, not when it is loaded.